### ISSUE TITLE
Type guard on `Number.isInteger` is too strong

### ISSUE THREAD

**Issue by ehoogeveen-medweb** [uhyo/better-typescript-lib#14]
I was reading the discussion at https://github.com/microsoft/TypeScript/issues/15048 and noticed that the definition of `Number.isInteger` from this library has the mentioned type guard, `number is number`.

To summarize the problem: When `Number.isInteger` returns `false`, the passed value may still be a `number` (just not an integer).

Unfortunately as that discussion shows there is currently no good way to represent this kind of 'weak type guard' in TypeScript, although https://github.com/microsoft/TypeScript/issues/15048#issuecomment-1174371299 does present an interesting workaround.

Adapted from that comment, you could do something like the following:
```typescript
declare const container: unique symbol;
type SubtypeOf<T> = T & { [container]: T };

interface NumberConstructor {
  isInteger<T>(number: T): number is number & SubtypeOf<T>;
}

function test(n: unknown) {
  if (typeof n === "string" || typeof n === "number") {
    const a = n; // string | number

    if (Number.isInteger(n)) {
      const b: number = n; // number & { [container]: string | number }
    } else {
      const e = n; // string | number
    }
  }
}
```
Although the truthy branch has an awkward type for `n`, the assignment to `b` is valid (and the falsy branch maintains the unconstrained type).

I don't know what the best solution is here - remove the type guard, apply the workaround, or leave the type guard in place. The current type guard is technically unsafe though.

**Comment by ehoogeveen-medweb** [comment]
Another workaround from that issue (my Typescript-fu isn't really good enough to appreciate the differences):

```typescript
declare abstract class OneSided {
  private static readonly $: unique symbol;
  private readonly [OneSided.$]: true;
}
type MakeOneSided<T> = T & OneSided;

interface NumberConstructor {
  isInteger<T>(number: T): number is number & MakeOneSided<T>;
}

function test(n: unknown) {
  if (typeof n === "string" || typeof n === "number") {
    const a = n; // string | number

    if (Number.isInteger(n)) {
      const b: number = n; // number & OneSided
    } else {
      const e = n; // string | number
    }
  }
}
```

**Comment by uhyo** [comment]
Thank you, this is fixed in v2.2.0 by removing the type guards.
